### PR TITLE
EVG-8234 GRPC client hang workaround: added timeout to err flush

### PR DESCRIPTION
**Pull request by hhoke**

### CODE REVIEW

File: agent/agent.go [hunk 1]
```diff
@@ -503,6 +503,8 @@ func (a *Agent) finishTask(ctx context.Context, tc *taskContext, status string)
 		err := a.uploadToS3(ctx, tc)
 		tc.logger.Execution().Error(errors.Wrap(err, "error uploading log files"))
 		tc.logger.Execution().Infof("Sending final status as: %v", detail.Status)
+		ctx, cancel = a.withCallbackTimeout(ctx, tc)
```
Comment (bsamek) [inline agent/agent.go:506]: This seems like kind of a long while to wait for flogs to flush. I think this defaults to 15 minutes.

Comment (hhoke) [reply agent/agent.go:506]: I think it's 15 minutes, or a project-defined timeout. I can change it to 1 minute?

File: agent/agent.go [hunk 2]
```diff
@@ -503,7 +503,9 @@ func (a *Agent) finishTask(ctx context.Context, tc *taskContext, status string)
 		err := a.uploadToS3(ctx, tc)
 		tc.logger.Execution().Error(errors.Wrap(err, "error uploading log files"))
 		tc.logger.Execution().Infof("Sending final status as: %v", detail.Status)
-		grip.Error(tc.logger.Flush(ctx))
+		flush_ctx, cancel := context.WithTimeout(ctx, 1*time.Minute)
```
Comment (bsamek) [inline agent/agent.go:506]: `time.Minute`, no need to multiply by 1 😄

Comment (hhoke) [reply agent/agent.go]: I remove

**Pull request merged**